test/generators/ModelGenerators.scala

```diff
@@ -129,6 +129,17 @@ trait ModelGenerators {
       Gen.oneOf(Locked, Unlocked, LockCheckFailure)
     }
 
+  lazy val submissionState: Arbitrary[SubmissionState] = Arbitrary {
```

Comment:
Could make this one `implicit` as this truly is an arbitrary submission state, also consider naming it `arbitrarySubmissionState`